extensions/ql-vscode/src/model-editor/modeling-store.ts

```diff
@@ -129,6 +162,100 @@ export class ModelingStore extends DisposableObject {
     });
   }
 
+  public addModeledMethods(
+    dbItem: DatabaseItem,
+    methods: Record<string, ModeledMethod>,
+  ) {
+    const state = this.getState(dbItem);
+
+    const newModeledMethods = {
+      ...methods,
+      ...Object.fromEntries(
+        Object.entries(state.modeledMethods).filter(
+          ([_, value]) => value.type !== "none",
+        ),
+      ),
+    };
+
+    this.onModeledMethodsChangedEventEmitter.fire({
+      modeledMethods: newModeledMethods,
+      dbUri: dbItem.databaseUri.toString(),
+      isActiveDb: dbItem.databaseUri.toString() === this.activeDb,
+    });
+  }
+
+  public setModeledMethods(
+    dbItem: DatabaseItem,
+    methods: Record<string, ModeledMethod>,
+  ) {
+    const state = this.getState(dbItem);
+    state.modeledMethods = methods;
+
+    this.onModeledMethodsChangedEventEmitter.fire({
+      modeledMethods: methods,
+      dbUri: dbItem.databaseUri.toString(),
+      isActiveDb: dbItem.databaseUri.toString() === this.activeDb,
+    });
+  }
+
+  public updateModeledMethod(dbItem: DatabaseItem, method: ModeledMethod) {
+    const state = this.getState(dbItem);
+    const methods = { ...state.modeledMethods, [method.signature]: method };
+    this.setModeledMethods(dbItem, methods);
+  }
+
+  public setModifiedMethods(
+    dbItem: DatabaseItem,
+    methodSignatures: Set<string>,
+  ) {
+    const state = this.getState(dbItem);
+    state.modifiedMethodSignatures = methodSignatures;
+
+    this.onModifiedMethodsChangedEventEmitter.fire({
+      modifiedMethods: state.modifiedMethodSignatures,
+      dbUri: dbItem.databaseUri.toString(),
+      isActiveDb: dbItem.databaseUri.toString() === this.activeDb,
+    });
+  }
+
+  public addModifiedMethods(
+    dbItem: DatabaseItem,
+    methodSignatures: Set<string>,
+  ) {
+    const state = this.getState(dbItem);
+
+    methodSignatures.forEach((signature) => {
+      state.modifiedMethodSignatures.add(signature);
+    });
+
+    this.onModifiedMethodsChangedEventEmitter.fire({
+      modifiedMethods: state.modifiedMethodSignatures,
+      dbUri: dbItem.databaseUri.toString(),
+      isActiveDb: dbItem.databaseUri.toString() === this.activeDb,
+    });
+  }
+
+  public addModifiedMethod(dbItem: DatabaseItem, methodSignature: string) {
+    this.addModifiedMethods(dbItem, new Set<string>([methodSignature]));
```

Comment:
Could we make the argument to `addModifiedMethods` an `Iterable<string>` instead so we don't need to construct a `Set`?